src/org/aion/abigenerator/ABICompilerMethodVisitor.java

```diff
@@ -46,13 +45,19 @@ public ABICompilerMethodVisitor(int access, String methodName, String methodDesc
     @Override
     public AnnotationVisitor visitAnnotation(String descriptor, boolean visible) {
         boolean isPublic = (this.access & Opcodes.ACC_PUBLIC) != 0;
-        if (!isPublic && Type.getType(descriptor).getClassName().equals(Callable.class.getName())) {
-            throw new CallableMismatchException("Annotation 'Callable' mismatches non-public access modifiers(protected/private)!");
-        }
-        if (isPublic && Type.getType(descriptor).getClassName().equals(Callable.class.getName())) {
+        boolean isStatic = (this.access & Opcodes.ACC_STATIC) != 0;
+        if(Type.getType(descriptor).getClassName().equals(Callable.class.getName()) ) {
+            if (!isPublic) {
+                throw new CallableMismatchNonPublicException("Annotation 'Callable' mismatches non-public access modifiers(protected/private)!");
+            }
+            if (!isStatic) {
+                throw new CallableMismatchNonStaticException("Annotation 'Callable' mismatches non-static access modifiers!");
```

Comment:
I think these messages should be hard-coded into their respective exception classes. I would pass the `methodName` to the Exception constructor, though, and include that as part of the message.

Reply:
Great comments.